lib/_stream_readable.js

```diff
@@ -490,10 +490,14 @@ Readable.prototype.read = function(n) {
     debug('length less than watermark', doRead);
   }
 
+  // If the readable side of a duplex stream has an error scheduled
+  // we should not read any further.
+  const errored = this._writableState && this._writableState.errored;
```

Comment:
Using the callback flow results in one extra tick which can cause an extra read to be performed. Luckily we now can inspect whether an error is scheduled on the writable side of a duplex stream.

Reply:
We need this since the callback error flow for Writable will always run async which means that a read can be done between `_write` failing and `destroy` being called.

Reply:
This is only relevant when `autoDestroy: false`.

Reply:
I'm not sure about this change. I would prefer not to have references to `_writableState` in `Readable`. I'm ok in adding `errored` to `_readableState` as it's long overdue.